catkin_ws/src/planner/src/substates/linear_search.py

```diff
@@ -6,35 +6,33 @@
 
 #ASSUMES AUV IS FACING DIRECTION TO SEARCH IN
 class LinearSearch(smach.State):
-    def __init__(self, timeout, forward_speed, control, mapping, target_class, min_objects):
+    def __init__(self, timeout, forward_speed, control, mapping, target_class, min_objects, search_depth):
         super().__init__(outcomes=['success', 'failure'])
         self.control = control
         self.mapping = mapping
         self.detectedObject = False
         self.target_class = target_class
         self.min_objects = min_objects
         self.timeout = timeout
+        self.search_depth = search_depth
         self.forward_speed = forward_speed
 
     def execute(self, ud):
         print("Starting linear search.")
-        #MOVE TO MIDDLE OF POOL DEPTH AND FLAT ORIENTATION
-        #[COMP] set to appropriate search depth
-        self.control.move((None, None, -1))
+        self.control.move((None, None, self.search_depth))
         self.control.rotateEuler((0,0,None))
```

Comment:
would it help if I made a quaternion based function called flatten

Reply:
Not in my opinion, what does it matter if its quaternion or not? Either way you will have to grab the yaw component through euler angles as well no?

Reply:
No it can be done with quaternions without euler angles. eulers might work but i'm not certain

Reply:
Alright sure we can do that, if you think it'll be more reliable I'm down. Since its out of scope of this PR though I'll add it as a task and merge this one once felipe has been able to test it

Reply:
https://github.com/mcgill-robotics/AUV-2024/issues/246